superset-frontend/packages/superset-ui-core/src/color/CategoricalColorNamespace.ts

```diff
@@ -39,8 +39,19 @@ export default class CategoricalColorNamespace {
 
   getScale(schemeId?: string) {
     const id = schemeId ?? getCategoricalSchemeRegistry().getDefaultKey() ?? '';
+    const scale = this.scales[id];
+    if (scale) {
+      return scale;
+    }
```

Comment:
I believe this might bring a few problems back. For instance, by reusing an existing scale like this in a dashboard, the charts would change colors based on the number of charts and their rendering order. We need to avoid that. Each chart should always use its own color scale and never share an existing color scale with any other chart within the same namespace and scheme. 

This is what makes the whole implementation much harder and the core issue I was mentioning. As discussed, one way to solve this problem might be to move the logic of the color generation in the Dashboard, by pulling all the labels of all the charts and only re-apply different colors to those labels that are shared across multiple charts, rather than sharing their scale.

Reply:
Hey @geido Thanks for the input!  This change was added temporarily and is not a major change, so I will restore it here. The major idea of this PR is to use a `default color scale` to save all the labels in the dashboard and regenerate them when the color schema is switched, as you said, but I try to keep the changes as small as possible here, there may be some problems, and I am keeping the test.

Reply:
I get all the labels by the `defaultColorScale.domain()`, then regenerate new colors when color schema changed and save them in the json metadata.